docs/apis/subsystems/tag/index.md

```diff
@@ -0,0 +1,274 @@
+
+This page describes API in Moodle 3.1 and above, for earlier versions see [Tag API before 3.1](https://docs.moodle.org/dev/Tag_API_before_3.1)
+
+## Tag API usage
+
+When a user tags something a **tag instance** is created in the database linking the item to the actual **tag**. If the tag did not exist before it is created automatically. Do not confuse these two entities - deleting the tag instance does not normally delete tag, however deleting tag deletes all tag instances associated with it.
+
+Developers define **tag areas** the areas that can be tagged, examples are:
+
+- blog posts
+- courses
+- users (tags represent user interests)
+- activity modules
+- questions
+- wiki pages
+
+Each tag area is identified by two attributes - component and itemtype. *Itemtype must be a name of a table in the database.* Component is the core component or plugin responsible for the tagging. This way the same DB table (for example 'user' or 'course') may be independently tagged by several components. Administrator or manager is able to manage the tag areas, collections and tags inside them on the [Managing tags](https://docs.moodle.org/en/Managing_tags) page. Users are able to search tags and view all items tagged with them that they have access to.
+
+### Defining a tag area
+
+First, developer must define the tag areas in the file **db/tag.php**. This will usually look like:
+
+```php
+$tagareas = array(
+    array(
+        'itemtype' => 'wiki_pages',  // This must be a name of the database table (without prefix).
+        'component' => 'mod_wiki', // This can be omitted for plugins since it can only be the full frankenstyle name of the plugin.
+        'callback' => 'mod_wiki_get_tagged_pages',
+        'callbackfile' => '/mod/wiki/locallib.php',
+    ),
+);
+```
+
+You will also need to add language string, for the example above it will be **$string[]('tagarea_wiki_pages') = 'Wiki pages';**
+
+There are more options such as specifying the default value for "Standard tags", having a fixed collection or excluding from search. They can be found in comments in [//github.com/moodle/moodle/blob/master/lib/db/tag.php lib/db/tag.php](https://docs.moodle.org/https///github.com/moodle/moodle/blob/master/lib/db/tag.php_lib/db/tag.php)
+
+After making changes to db/tag.php plugin developer must bump the plugin version in **version.php** and run upgrade script. This usually applies to any changes in the db/ folder.
```

Comment:
```suggestion

:::important

After making changes to db/tag.php, you must bump the plugin version in **version.php** and perform a Moodle upgrade.

:::
```

Note: I'd also suggest that we create an entry in https://moodledev.io/docs/apis/commonfiles
And ideally move the documentation linked to in the previous paragraph into there (see version.php as an example).